docs/content/index.md

```diff
@@ -2,60 +2,153 @@
+     timestamp             publisher          advertiser  gender country impressions clicks revenue
+     2011-01-01T01:00:00Z  ultratrimfast.com  google.com  Male   USA     1800        25     15.70
+     2011-01-01T01:00:00Z  bieberfever.com    google.com  Male   USA     2912        42     29.18
+     2011-01-01T02:00:00Z  ultratrimfast.com  google.com  Male   UK      1953        17     17.31
+     2011-01-01T02:00:00Z  bieberfever.com    google.com  Male   UK      3194        170    34.01
 
-Architecture Overview
----------------------
+In practice, we see that rolling up data can dramatically reduce the size of data that needs to be stored (often by a factor of 100).
+This storage reduction does come at a cost; as we roll up data, we lose the ability to query individual events. Phrased another way,
+the rollup granularity is the minimum granularity you will be able to query data at. Hence, Druid ingestion specs define this granularity as the `queryGranularity` of the data.
+
+Druid is designed to perform single table operations and does not currently support joins.
+Although many production setups instrument joins at the ETL level, data must be denormalized before it is loaded into Druid.
+
+## Indexing the Data
+
+Druid gets its speed in part from how it stores data. Borrowing ideas from search infrastructure,
+Druid creates immutable snapshots of data, stored in data structures highly optimized for analytic queries.
+
+Druid is a column store, which means each individual column is stored separately. Only the columns that pertain to a query are used
+in that query, and Druid is pretty good about only scanning exactly what it needs for a query.
+Different columns can also employ different compression methods. Different columns can also have different indexes associated with them.
+
+## Sharding the Data
+
+Druid shards are called `segments` and Druid always first shards data by time. In our compacted data set, we can create two segments, one for each hour of data.
+
+For example:
+
+Segment `sampleData_2011-01-01T01:00:00:00Z_2011-01-01T02:00:00:00Z_v1_0` contains
+
+     2011-01-01T01:00:00Z  ultratrimfast.com  google.com  Male   USA     1800        25     15.70
+     2011-01-01T01:00:00Z  bieberfever.com    google.com  Male   USA     2912        42     29.18
+
+
+Segment `sampleData_2011-01-01T02:00:00:00Z_2011-01-01T03:00:00:00Z_v1_0` contains
+
+     2011-01-01T02:00:00Z  ultratrimfast.com  google.com  Male   UK      1953        17     17.31
+     2011-01-01T02:00:00Z  bieberfever.com    google.com  Male   UK      3194        170    34.01
+
+Segments are self-contained containers for the time interval of data they hold. Segments
+contain data stored in compressed column orientations, along with the indexes for those columns. Druid queries only understand how to
+scan segments.
+
+Segments are uniquely identified by a datasource, interval, version, and partition number.
```

Comment:
I think the stuff about blocks can be taken out or at least deferred until Design, for a couple reasons.

- The treatment here is not really correct. Some blocks are queryable without being complete (linear shard specs, possibly a new elastic shard spec)
- At this point I think we mostly want to give people a flavor of how data is stored in Druid, so talking about segments is enough. The block and timeline stuff isn't really adding much to the flavor.